### DISCUSSION TITLE
Epiverse/WHO workshop feedback

### DISCUSSION THREAD

**Discussion by adamkucharski** [epicentre-msf/epishiny#9] [General]
This looks like a great resource! We're documenting some initial rough exploration on a fork here: https://github.com/epiverse-trace/epishiny/issues

Will review and summarise ideas this afternoon, then hopefully convert to issues/PRs where relevant on main epishiny repo.

**Comment by PaulC91** [comment]
Hi all, I'm Paul the package dev of epishiny at Epicentre.

Sorry for the initial installation and dependency issues you had, the package wasn't quite ready for a release but hopefully is working ok now with a fresh installation.

I've added documentation to all exported functions with examples of how to run inside a shiny app.

Let me know here if you have any questions about the package or if you run into any more issues when testing it out and I'll try to respond ASAP.

Cheers

**Reply by adamkucharski** [reply]
Yep, installation all working smoothly now! Thanks! Will be in touch if have questions, or later once have more complete ideas.

**Comment by adamkucharski** [comment]
We've added some pull requests documenting ideas/suggestions that have emerged during our package exploration over past day or so (collaborative work with @sbfnk @martinamcm @rebeccanash @mv2405). Brief summary:

- Vignette on incorporation of early data: https://github.com/epiverse-trace/epishiny/commit/854ca53540e5149764add5ffc79fa1581413a7c1
- PR to improve joining, inspired by above task: https://github.com/epicentre-msf/epishiny/pull/12
- PR to improve use of naming, inspired by above task: https://github.com/epicentre-msf/epishiny/pull/11
- Draft PR with vignette on adding new module as a new user, and several potential issue suggestions identified along the way: https://github.com/epicentre-msf/epishiny/pull/13
- Draft PR with vignette giving overview of package as seen by new user, additional function notation, and suggested refinements to visualisation functionality and CFR estimation: https://github.com/epicentre-msf/epishiny/pull/14

One thing in particular that came out of discussions is the scope of the package for providing narrower opinionated templates in central package structure with functionality common across all users vs allowing users more flexibility to define own modules/options to create customised local apps that could then be deployed.

**Reply by sbfnk** [reply]
> Vignette on incorporation of early data: https://github.com/epiverse-trace/epishiny/commit/854ca53540e5149764add5ffc79fa1581413a7c1

minor correction that it's on *external* (not necessarily early) data

**Reply by PaulC91** [reply]
Thanks a lot for all the contributions! I'll review them over the coming days and try to incorporate at much as possible. I'll keep you updated on progress in this discussion. Cheers!

**Comment by pratikunterwegs** [comment]
Hi @PaulC91 just jumping on to this discussion thread - thanks for organising the meet on Tuesday. I'm happy to help out with features that UKHSA et al might want to add (choropleths, aggregated data for modules). At Epiverse I'm also working with Adam to push out a [package for CFR estimation while correcting for delays](https://github.com/epiverse-trace/cfr) (static, rolling or time-varying), and that could be a good module to add - let me know once the contributing guidance is ready and I could mock up a PR? Happy to hear thoughts on how to make this as useful as possible to intended users - including potential edits to {cfr}.

**Reply by PaulC91** [reply]
Hi @pratikunterwegs,

Sorry for the slow reply I missed you message! Thanks a lot for your interest in contributing. I added a quick [contributing guide](https://epicentre-msf.github.io/epishiny/articles/contributing.html) that I see is pretty similar to the one in {cfr} so I'm sure you won't have any issues! 

I recently added some choropleth functionality to the place module that maps attack rates provided a `pop_var` variable name is passed as part of the `geo_data` specification. You can check `launch_demo_dashboard()` to see it in action. Please feel free to suggest improvements to this via issues or PRs.

Any help on allowing for aggregated datasets in the package would be very welcome also as well as a {cfr} module!

Looking forward to hear your ideas.

Paul

**Reply by pratikunterwegs** [reply]
Thanks @PaulC91 will defintely take a look at the new functionality and make any suggestions I can - I do love making maps.

Will make a fork to see how I can include CFR functionality, ideally using the [{cfr} package](https://github.com/epiverse-trace/cfr) I hope to release soon. The package works with aggregated data, so might take some thought. Happy to assist or coordinate with others building support the aggregated-data functionality. cc-ing @adamkucharski as well to keep updated.
